pkg/execution/state/redis_state/queue_test.go

```diff
@@ -2028,6 +2080,80 @@ func TestShardLease(t *testing.T) {
 	})
 }
 
+// TestQueueRateLimit asserts that the queue respects rate limits when added to a queue item.
+func TestQueueRateLimit(t *testing.T) {
+	r := miniredis.RunT(t)
+	rc, err := rueidis.NewClient(rueidis.ClientOption{
+		InitAddress:  []string{r.Addr()},
+		DisableCache: true,
+	})
+	require.NoError(t, err)
+	defer rc.Close()
+	ctx := context.Background()
+	q := NewQueue(rc)
+
+	idA, idB := uuid.New(), uuid.New()
+
+	_ = idB
+
+	t.Run("Without bursts", func(t *testing.T) {
+		throttle := &osqueue.Throttle{
+			Key:    "some-key",
+			Limit:  1,
```

Comment:
could you add another test here that's
```
Throttle{
  Key: "some-other-key",
  Limit: 1,
  Period: 5,
  Burst: 2
}
```
Would be good to verify that bursting is working as expected (in this case 3 should pass), since the current test has no bursts.

Reply:
Added, thanks!